src/AppBundle/Controller/DashboardController.php

```diff
@@ -0,0 +1,349 @@
+            'dashboard' => $dashboard
+        ));
+    }
+
+    //.......Delete dashboard
+    /***
+     * @param $dashboard
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function deleteDashAction(Dashboard $dashboard)
+    {
+        //.......Get current dashboard
+        $em = $this->getDoctrine()->getManager();
+        $id = $dashboard->getId();
+        //.......get the components that are on the dashboard
+        $components = $em->getRepository('AppBundle:Component')->findBy(
+            array('dashboard' => $id));
+
+        //.......delete the components
+        //.......delete the dashboard and update database
+        if ($dashboard != null) {
+            foreach ($components as $component) {
+                $em->remove($component);
+            }
+            $em->remove($dashboard);
+            $em->flush();
+        }
+
+        return $this->redirectToRoute('app_home', array(
+            'id' => $id));
+    }
+
+    //.......Leave dashboard function - current user can quit a dashboard where he's been added
+    /***
+     * @param $dashboard
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function leaveDashAction(Dashboard $dashboard)
+    {
+        //.......get current user
+        $user = $this->container->get('security.context')->getToken()->getUser();
+        $em = $this->getDoctrine()->getManager();
+        //.......get current dashboard
+        $id = $dashboard->getId();
+        //.......remove current user from dashboard
+        $dashboard->removeCollaborator($user);
+        $em->flush();
+
+        return $this->redirectToRoute('app_home', array(
+            'id' => $id));//to do redirect to homepage
+    }
+
+    //.......Delete Component
+    /***
+     * @param $componentId
+     * @param $id
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function deleteComponentAction($componentId, $id)
+    {
+        //......get component by id
+        $em = $this->getDoctrine()->getManager();
+        $component = $em->getRepository('AppBundle:Component')->find($componentId);
+        //......delete component
+        if ($component != null) {
+            $em->remove($component);
+            $em->flush();
+        }
+
+        return $this->redirectToRoute('app_dashboard', array(
+            'id' => $id
+        ));
+    }
+    //.......Change graph size
+    /***
+     * @param $componentId
+     * @param $size
+     * @param $id
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function changeSizeAction($componentId, $size, $id)
+    {
+        //......get current component
+        $em = $this->getDoctrine()->getManager();
+        $component = $em->getRepository('AppBundle:Component')->find($componentId);
+        //......set component size
+        $component->setSizeComponent($size);
+        $em->persist($component);
```

Comment:
You don't need to persist the component, it already exists in the database,, just flush changes